File: app/src/main/java/org/bitcoindotcom/bchprocessor/bip70/Bip70PayService.java
package org.bitcoindotcom.bchprocessor.bip70;

import org.bitcoindotcom.bchprocessor.bip70.model.InvoiceCreation;
import org.bitcoindotcom.bchprocessor.bip70.model.InvoiceStatus;

import retrofit2.Call;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;
import retrofit2.http.Body;
import retrofit2.http.POST;

public interface Bip70PayService {
    static Bip70PayService create(String baseUrl) {
        Retrofit retrofit = new Retrofit.Builder()
                .baseUrl(baseUrl)
                .addConverterFactory(GsonConverterFactory.create())
                .build();
        return retrofit.create(Bip70PayService.class);
    }

    @POST("create_invoice")
    Call<InvoiceStatus> createInvoice(@Body InvoiceCreation r);
}

File: app/src/main/java/org/bitcoindotcom/bchprocessor/bip70/Bip70PayService.kt
package org.bitcoindotcom.bchprocessor.bip70

import com.bitcoin.merchant.app.util.GsonUtil
import org.bitcoindotcom.bchprocessor.bip70.model.InvoiceRequest
import org.bitcoindotcom.bchprocessor.bip70.model.InvoiceStatus
import retrofit2.Call
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory
import retrofit2.http.Body
import retrofit2.http.POST

interface Bip70PayService {
    @POST("create_invoice")
    fun createInvoice(@Body r: InvoiceRequest): Call<InvoiceStatus?>

    companion object {
        fun create(baseUrl: String): Bip70PayService {
            return Retrofit.Builder()
                    .baseUrl(baseUrl)
                    .addConverterFactory(GsonConverterFactory.create(GsonUtil.gson))
                    .build()
                    .create(Bip70PayService::class.java)
        }
    }
}